app/controllers/application_controller.rb

```diff
@@ -12,6 +12,17 @@ class ApplicationController < ActionController::Base
 
 protected
 
+  def get_payload(jwt)
+    if jwt
+      payload = ApplicationKey.validate_jwt!(jwt)
+      create_jwt(payload)
```

Comment:
Move the definition of this method to `ApplicationController`, as it's called in the `WelcomeController`, `DeviseRegistrationsController`, and `DeviseSessionsController` (and is only defined in the latter two).

Or even just inline the method, it's pretty small.

Reply:
Also it probably makes sense to move the `validate_jwt!` method from `ApplicationKey` to `Jwt`

Reply:
That seems like a good idea.  I've added it as a `before_save` callback.